libretro/libretro-core.c

```diff
@@ -1417,34 +1454,187 @@ void retro_set_environment(retro_environment_t cb)
          "enabled"
       },
 #endif
+
+
 #if defined(__X64__) || defined(__X64SC__) || defined(__X128__) || defined(__VIC20__) || defined(__PLUS4__)
       {
-         "vice_aspect_ratio",
-         "Aspect Ratio",
+         "vice_bordercrop_horiz",       // Horizontal Cropping menu item & settings
+         "Crop Horizontal Borders",
          "",
          {
-            { "auto", "Automatic" },
-            { "pal", "PAL" },
-            { "ntsc", "NTSC" },
-            { "raw", "1:1" },
+            { "none", "disabled" },
+            { "small", "Small Crop" },
+            { "medium", "Medium Crop" },
+            { "full", "Full Crop" },
             { NULL, NULL },
          },
-         "auto"
+         "none"
       },
+
       {
-         "vice_zoom_mode",
-         "Zoom Mode",
-         "Zoom will be ignored without borders.\nRequirements in RetroArch settings:\n- Aspect Ratio: Core provided,\n- Integer Scale: Off.",
+         "vice_zoom_mode",              // Vertical Cropping menu item & settings - option name "vice_zoom_mode" for legacy compatibility with existing options files
+         "Crop Vertical Borders",
+         "Crop only when Display Borders 'ON'\nRetroArch Video/Scaling Settings required: Aspect Ratio: Core provided, Integer Scale: Off",
          {
             { "none", "disabled" },
-            { "small", "Small" },
-            { "medium", "Medium" },
-            { "maximum", "Maximum" },
+            { "small", "Small Crop" },
+            { "medium", "Medium Crop" },
+            { "full", "Full Crop" },
+  #if defined(__X64__) || defined(__X64SC__) || defined(__X128__)   // only include manual cropping for c64/128
+           { "manual", "Manual Crop" },
+  #endif
             { NULL, NULL },
          },
          "none"
       },
+
+  #if defined(__X64__) || defined(__X64SC__) || defined(__X128__)
+      {
+         "vice_bordercrop_top",         // Manual Pixel settings for Top Cropping option
+         "Manual Top Cropping",

```

Comment:
Why is pixel selection needed as a separate option? Is small, medium, full not enough? If you really need pixel values, maybe add them to the other choices of "small, medium, full" e.g. small, medium, full, 0 px, 1px, so the pixel values just appear as additional choices to the crop option.

Reply:
Also maybe just give a few pixel values not every. E.g. 2, 4, 6, 8, etc.

Reply:
I did have it every 2 in development - but RA's menu selection is so fast using the new UI, that there's nothing inherently wrong with individual pixels.
The precision is necessary to accommodate overrides for specific content.  These settings have now been tested with hundreds of games.

Reply:
Ok I still think these should be added to existing zoom options just as extra selections to keep it simple.

Reply:
Please don't evaluate these changes without first having tried the code live. Check-out my fork compile it and try it. There's nothing complicated about the presentation nor use. It was created and refined and tested to be simple to understand, concise and quick to change.

Reply:
Another reason they're separate is to not complicate the Vertical options list. It's nice and simple at 4 presets.  If you only want to use presets, then you can do so and not be bothered to look at any pixels.  These menus are closed and start at 0. You have to purposefully open them or change the values to use.  No complication needed.